Voice Note/ViewModels/TemperatureDataService.swift

```diff
@@ -0,0 +1,80 @@
+// 
+// Created by Anwar Ulhaq on 25.4.2023.
+// 
+// 
+
+import Foundation
+
+class TemperatureDataService: ObservableObject {
+    @Published private(set) var temperatures: [Temperature] = []
+    @Published private(set) var isLoading = false
+
+    private var request: TemperatureAPIRequest<TemperatureResource>?
+
+    // TODO should call fetchLatestTemperatures init
+    private func loadLocationsFromCoreData() -> [Location] {
+        CoreDataService.localStorage.fetchAllVoiceNotes().map { voiceNote -> Location? in
+                    voiceNote.location
+                }
+                .compactMap {$0}
+    }
+
+    func fetchLatestTemperatures() async {
+
+        _ = await withTaskGroup(of: (Temperature?, VoiceNote?).self) { group in
+            for voiceNote in CoreDataService.localStorage.fetchAllVoiceNotes() {
+                if let location = voiceNote.location {
+                    group.addTask { [self] in
+                        async let temps = try? await loadCurrentTemperature(atLocation: location)
+                        return (await temps, voiceNote)
+                    }
+                }
+            }
+
+            let context = CoreDataService.localStorage.getManageObjectContext()
+            var latestTemps: [(Temperature, VoiceNote)] = []
+
+            for await result in group {
+                if let newTemperature = result.0, let voiceNote = result.1 {
+                    voiceNote.weather?.temperature = newTemperature
+                    do {
+                        try context.save()
+                    } catch {
+                        // TODO handle error
+                    }
+                }
+            }
+            return latestTemps
+        }
+    }
+
+    func loadSevenDaysTemperature(atLocation: Location) async throws -> [Temperature] {
```

Comment:
Great work @anwar-ulhaq,  I have viewed your code and so far, I am not quite sure whether we need this loadSevenDaysTemperature() here. User can view the specific location temperature when clicking on the map pin? So I think this function is not necessary to have in our app. The rest of functions in this class seem reasonable to me except for this one

Reply:
We can omit this code, for our app we can simply show the current temperature.

Reply:
The API that i use returns next seven days temperature in an array for a single location. So for the sake of  code readability, i refactor that piece of code in a separate function. I can not remove this code. I can make this function private though, but i think i left it as default internal in view that we might need it a some point to get next seven day forecast at VoiceNote location.